marda_registry/models/__init__.py

```diff
@@ -3,9 +3,52 @@
 be edited by hand.
 """
 
-__version__ = "0.0.1"
+from pydantic import validator
 
-from .extractor import Extractor
+from .extractor import Extractor as AutoGenExtractor
 from .filetype import FileType
 
+
+class Extractor(AutoGenExtractor):
+    """A wrapper for the automatically generated Extractor class
+    that includes some additional validators.
+
+    """
+
+    @validator("usage")
+    def check_usage_filetypes(cls, v, values):
+        """Check that any filetype instructions in the usage
+        are defined at the top-level, and that all supported_filetypes
+        values have instructions (or that there is a catch-all).
+
+        """
+
+        if not v:
+            return v
+
+        supported_filetypes = values.get("supported_filetypes")
+        usage_to_fts = {}
+
+        bad_fts = set()
+        if usage := v:
```

Comment:
You could just do:
```
if not v:
    return v
else:
    usage = v
```
and avoid the fancy `:=` operator that I had to google. It's not like evaluating `v` is expensive...